build/scripts/Products.fsx

```diff
@@ -162,6 +175,14 @@ module Products =
                                 if directoryExists original |> not then
                                     Rename (InDir @@ (sprintf "kibana-%s" version.FullVersion)) (InDir @@ original)
                             | _ -> ()
+                        
+                        // Snapshots need renaming as folder inside zip is named differently
+                        let useSnapshots = getBuildParamOrDefault "snapshots" "$false"
+                        if (useSnapshots = "$true") then                                              
+                            let existing = InDir @@ (sprintf "%s-%s" product.Name (Snapshots.GetVersions |> Seq.head))
+                            let target = InDir @@ (sprintf "%s-%s" product.Name version.FullVersion)
+                            Rename target existing
```

Comment:
Using an example:

The hosted zip filename is: 
`https://snapshots.elastic.co/7.0.0-alpha1-ea57ee52/downloads/elasticsearch/elasticsearch-7.0.0-alpha1-SNAPSHOT.zip`

This is downloaded locally into the `in` dir as: `elasticsearch-7.0.0-alpha1-ea57ee52.zip`

When extracted it creates a folder called: `elasticsearch-7.0.0-alpha1-snapshot`

This is then renamed to `elasticsearch-7.0.0-alpha1-ea57ee52`

When you re-run the build script with the same parameters (assuming it all downloaded and built successfully) then it will simply re-use the extracted folder, and if that is missing, will re-use the downloaded zip file.

In short, it caches as expected.

Reply:
:+1: